rust/kernel/file_operations.rs

```diff
@@ -240,41 +269,46 @@ pub trait FileOperations: Sync + Sized {
 
     /// Cleans up after the last reference to the file goes away.
     ///
-    /// Note that the object is moved, so it will be freed automatically unless
-    /// the implemention moves it elsewhere.
+    /// Note that the object is moved, so it will be freed automatically unless the implemention
+    /// moves it elsewhere.
     ///
-    /// Corresponds to the `release` function pointer in
-    /// `struct file_operations`.
+    /// Corresponds to the `release` function pointer in `struct file_operations`.
     fn release(_obj: Self::Wrapper, _file: &File) {}
 
     /// Reads data from this file to userspace.
     ///
     /// Corresponds to the `read` function pointer in `struct file_operations`.
-    const READ: ReadFn<Self> = None;
+    fn read(&self, _file: &File, _data: &mut UserSlicePtrWriter, _offset: u64) -> KernelResult<()> {
+        Err(Error::EINVAL)
```

Comment:
These could all be `unreachable!()`, right? Or, I guess you'd reach them if you implemented the trait but didn't override the function?

Reply:
It's reachable if a type implements the trait, doesn't implement this function, and lists it in `declare_file_operations`.